lib/explorer/polars_backend/native.ex

```diff
@@ -69,6 +69,7 @@ defmodule Explorer.PolarsBackend.Native do
   def df_n_chunks(_df), do: err()
   def df_new(_cols), do: err()
   def df_pivot_wider(_df, _id_cols, _pivot_column, _values_columns), do: err()
+  def df_pretty_inspect(_df, _nrow), do: err()
```

Comment:
Is this a data frame callback? Instead shouldn't we have the whole printing in Elixir instead instead of native?

Reply:
For some reason I thought that we needed this callback, I'll remove it.
Sorry 😳

Reply:
No worries at all. Also note that Explorer.Inspect is private so we may want a public API for the inspect function. We may also want to filter the table somehow... printing the whole table can be too much? Should we automatically limit it 100 elements and have a `limit: ` flag?

Reply:
No worries at all. Also note that Explorer.Inspect is private so we may want a public API for the inspect function.

Reply:
Well, I'll need some help on how to do that. I'm new on that Elixir thing 😅

Reply:
@kimjoaoun I will be glad to help! The first question is what the public API should be and once @cigrainger lets us know his preference we will get it done. :)

Reply:
@josevalim do you mean re: how to inspect? I think the table view as the default `Inspect` implementation and then `Explorer.DataFrame.glimpse()` for the current (non-tabular) inspect implementation.

Reply:
Understood. So the current `Inspect` would become a normal function and the tabular goes as the `Inspect`.
Wouldn't this break all the doctests?

Reply:
Exactly. And yes, yes it would. :grimacing:

Reply:
I'm also happy to flip the other direction and add the tabular version as a normal function. That would be `Explorer.DataFrame.table()` or similar. We had had some discussions in which we figured more people would be comfortable with a table by default. I prefer the `glimpse` inspired version for a number of reasons. Perhaps we should start by adding the table as a normal function and see how we go?

Reply:
Well, I see no problem in doing all in this PR, but it'll be a long one with a lot of changes.
So the roadmap of this PR now is:

- [ ] Adding `nrow` and `ncol` to table printing.
- [ ] Replacing the current Inspect with the table printer
- [ ] Fixing all tests.

Reply:
A table function sounds like a great starting point!

Reply:
Ok!
I think I'll need to start a new PR, since I deleted my fork, but I forgot doing so I'm not be able to do changes in this PR.

Reply:
I agree, let's start with the table function in this PR and then reassess.